src/rpc/methods/f3.rs

```diff
@@ -430,15 +435,31 @@ impl RpcMethod<0> for GetParticipatingMinerIDs {
     type Params = ();
     type Ok = Vec<u64>;
 
-    async fn handle(ctx: Ctx<impl Blockstore>, _: Self::Params) -> Result<Self::Ok, ServerError> {
-        match ctx.chain_config().network {
-            NetworkChain::Devnet(_) => {
-                // For now, just hard code the devnet miner for testing
-                let shared_miner_addr = Address::from_str("t01000")?;
-                Ok(vec![shared_miner_addr.id()?])
+    async fn handle(_: Ctx<impl Blockstore>, _: Self::Params) -> Result<Self::Ok, ServerError> {
+        const ENV_KEY: &str = "FOREST_F3_PERMENANT_PARTICIPATING_MINER_ADDRESSES";
+        let mut ids: Vec<u64> = F3_PARTICIPANT_LEASES.read().keys().copied().collect();
+        if let Ok(permenant_addrs) = std::env::var(ENV_KEY) {
+            let mut extended_ids = HashSet::default();
+            for addr_str in permenant_addrs.split(",") {
+                let addr = Address::from_str(addr_str.trim())
+                    .with_context(|| {
+                        format!("Failed to parse miner address {addr_str} set in {ENV_KEY}")
+                    })
+                    .unwrap();
```

Comment:
Is unwrapping the correct behaviour? This would mean the node would crash an incorrect env variable. Isn't it too harsh?

Reply:
Unwraps have to turn into Result indeed.

Reply:
If I'm being completely honest, since those are specified via an ENV variable on node startup - I think the correct behaviour is to parse them on startup and fail fast if they don't parse.
Then later we just serve them from the storage, even if it's an in-memory store.

Reply:
Good suggestion. Fixed.